src/Npgsql/Internal/NpgsqlConnector.Auth.cs

```diff
@@ -57,7 +57,7 @@ async Task Authenticate(string username, NpgsqlTimeout timeout, bool async, Canc
     async Task AuthenticateCleartext(string username, bool async, CancellationToken cancellationToken = default)
     {
         var passwd = await GetPassword(username, async, cancellationToken).ConfigureAwait(false);
-        if (passwd == null)
+        if (string.IsNullOrEmpty(passwd))
```

Comment:
It seems like one of the tests somehow changes password from `null` to an empty string. No idea which one and why only now...